tracer/src/Datadog.Trace/Telemetry/Collectors/DependencyTelemetryCollector.cs

```diff
@@ -71,12 +69,27 @@ public bool HasChanges()
         public ICollection<DependencyTelemetryData> GetData()
         {
             var hasChanges = Interlocked.CompareExchange(ref _hasChangesFlag, 0, 1) == 1;
+
             if (!hasChanges)
             {
                 return null;
             }
 
-            return _assemblies.Keys;
+            var assembliesToRecord = new List<DependencyTelemetryData>();
+            foreach (var assembly in _assemblies)
+            {
+                if (assembly.Value == false)
+                {
+                    assembliesToRecord.Add(assembly.Key);
+                }
+            }
+
+            foreach (var key in assembliesToRecord)
+            {
+                _assemblies[key] = true;
+            }
```

Comment:
I don't think it's necessary to do two loops here?
```suggestion
            foreach (var assembly in _assemblies)
            {
                if (assembly.Value == false)
                {
                    _assemblies[assembly.Key] = true;
                    assembliesToRecord.Add(assembly.Key);
                }
            }
```

Reply:
It's against my very core to alter a collection when iterating over it. But sure I can do that

Reply:
I mean, I had to read it a couple of times to understand you _weren't_ doing that, but I agree 😛 _Pretty_ sure it's safe though, as this is guaranteed to update an existing key

Reply:
fixed in 2aa638d121e24ba0d014978b6339a4409fc3778b